modules/common.nix

```diff
@@ -93,14 +109,20 @@
         };
       };
 
-      config = {
-        target = mkDefault name;
-        source = mkIf (config.text != null) (mkDerivedConfig options.text (text:
-          pkgs.writeTextFile {
-            inherit name text;
-            inherit (config) executable;
-          }));
-      };
+      config = mkMerge [
+        {
+          target = mkDefault name;
+          source = mkIf (config.text != null) (mkDerivedConfig options.text (text:
+            pkgs.writeTextFile {
+              inherit name text;
+              inherit (config) executable;
+            }));
+        }
+
+        (lib.mkIf (config.generator != null) {
+          text = mkDefault (config.generator config.value);
+        })
```

Comment:
I was thinking of adding a `config.generator != null -> config.value != null` assertion here (or well, at the top level since assertions don't work correctly in submodules IIRC) to handle edge cases, but I'm not sure if it's that's super needed/wanted